finetuner/tuner/keras/__init__.py

```diff
@@ -62,60 +61,41 @@ def _get_default_optimizer(self, learning_rate: float) -> Optimizer:
 
         return tf.keras.optimizers.Adam(learning_rate=learning_rate)
 
-    def _train(
-        self, data: KerasSequenceAdapter, optimizer: Optimizer, description: str
-    ) -> ScalarSequence:
+    def _train(self, data: KerasSequenceAdapter):
         """Train the model on given labeled data"""
 
-        _summary = ScalarSequence('Train Loss')
-        with ProgressBar(
-            description,
-            message_on_done=_summary.__str__,
-            final_line_feed=False,
-            total_length=data.get_size(),
-        ) as p:
-            for inputs, labels in data.get_dataset():
-                with tf.GradientTape() as tape:
-                    embeddings = self._embed_model(inputs)
-                    loss = self._loss(embeddings, labels)
+        for idx, (inputs, labels) in enumerate(data.get_dataset()):
+            self.state.batch_index = idx
+            self._trigger_callbacks('on_train_batch_begin')
 
-                grads = tape.gradient(loss, self._embed_model.trainable_weights)
-                optimizer.apply_gradients(
-                    zip(grads, self._embed_model.trainable_weights)
-                )
+            with tf.GradientTape() as tape:
+                embeddings = self._embed_model(inputs)
+                loss = self._loss(embeddings, labels)
 
-                _summary += loss.numpy()
+            grads = tape.gradient(loss, self._embed_model.trainable_weights)
+            self._optimizer.apply_gradients(
+                zip(grads, self._embed_model.trainable_weights)
+            )
 
-                p.update(message=str(_summary))
+            self.state.current_loss = loss.numpy()
+            self._trigger_callbacks('on_train_batch_end')
 
         data.on_epoch_end()  # To re-create batches
-        return _summary
 
-    def _eval(
-        self,
-        data: KerasSequenceAdapter,
-        description: str = 'Evaluating',
-        train_loss: Optional[ScalarSequence] = None,
-    ) -> ScalarSequence:
+    def _eval(self, data: KerasSequenceAdapter):
         """Evaluate the model on given labeled data"""
 
-        _summary = ScalarSequence('Eval Loss')
-
-        with ProgressBar(
-            description,
-            message_on_done=lambda: f'{train_loss} | {_summary}',
-            total_length=data.get_size(),
-        ) as p:
-            for inputs, labels in data.get_dataset():
-                embeddings = self._embed_model(inputs)
-                loss = self._loss(embeddings, labels)
+        for idx, (inputs, labels) in enumerate(data.get_dataset()):
+            self.state.batch_index = idx
+            self._trigger_callbacks('on_val_batch_begin')
```

Comment:
if we didn't implement a certain method in a callback, it will go for a silent pass?

Reply:
Then it defaults to method of `BaseCallback` which is a silent pass